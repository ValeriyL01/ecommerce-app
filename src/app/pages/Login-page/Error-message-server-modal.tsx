import React, { useState } from 'react'

import BaseButton from 'components/shared/BaseButton/BaseButton'

interface ErrorModalProps {
  errorMessage: string
  onErrorMessageChange: (value: string) => void
}

export const ErrorModal: React.FC<ErrorModalProps> = ({ errorMessage, onErrorMessageChange }) => {
  const [isButtonClicked, setIsButtonClicked] = useState(false)

  const handleButtonClick = (): void => {
    setIsButtonClicked(true)
    onErrorMessageChange('')
  }
  return (
    <div
      className={`fixed left-0 top-0 z-20 flex h-full w-full items-center justify-center bg-gray-800 bg-opacity-80  transition duration-100 ${isButtonClicked ? 'hidden opacity-0' : 'opacity-1 visible'}`}
    >
      <div
        className={`  flex h-1/4 w-1/4  flex-col items-center justify-center gap-10 bg-[#783939] p-4 font-osvald text-xl text-white`}
      >
        <span className="">{errorMessage}</span>
        <BaseButton variant="basket" onClick={handleButtonClick}>
          Ok
        </BaseButton>
      </div>
    </div>
  )
}
