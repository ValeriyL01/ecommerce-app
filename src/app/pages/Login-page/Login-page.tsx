export const LoginPage: React.FC = () => {
  return <h1>This is Login Page</h1>
}
