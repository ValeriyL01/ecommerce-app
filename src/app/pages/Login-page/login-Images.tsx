export const LoginImages: React.FC = () => {
  return (
    <>
      <img
        className="absolute right-0 top-0 z-10 w-1/4 object-cover"
        src="/images/loginPageImg/loginPage-flower1.png"
        alt=""
        data-testid="loginFlower1"
      />
      <img
        className="absolute bottom-0 left-0 z-10 w-1/4 object-cover"
        src="/images/loginPageImg/loginPage-flower2.png"
        alt=""
        data-testid="loginFlower2"
      />
      <img
        className=" z-5 absolute right-0 top-0  w-1/4 object-cover"
        src="/images/loginPageImg/Ellipse rt.png"
        alt=""
        data-testid="loginEllipseRt"
      />
      <img
        className=" z-5 absolute bottom-0 left-0  w-1/4 object-cover"
        src="/images/loginPageImg/Ellipse lb.png"
        alt=""
        data-testid="loginEllipseLb"
      />
      <img
        className=" z-5 absolute left-0 top-0  w-2/12 object-cover"
        src="/images/loginPageImg/Ellipse lt.png"
        alt=""
        data-testid="loginEllipseLt"
      />
      <img
        className=" z-5 absolute bottom-0 right-0  w-2/12 object-cover"
        src="/images/loginPageImg/Ellipse rb.png"
        alt=""
        data-testid="loginEllipseRb"
      />
      <img
        className=" z-5 absolute left-1/2 top-1/2 w-[250px]  object-cover"
        src="/images/loginPageImg/signature.png"
        alt=""
        data-testid="loginSignature"
      />
    </>
  )
}
