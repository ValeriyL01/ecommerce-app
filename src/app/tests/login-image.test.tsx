import { render, screen } from '@testing-library/react'

import '@testing-library/jest-dom'
import { LoginImages } from 'pages/Login-page/login-Images'

/**
 * @jest-environment jsdom
 */

describe('Registration-images', () => {
  it('should render images', () => {
    render(<LoginImages />)

    const loginFlower1 = screen.getByTestId('loginFlower1')
    const loginFlower2 = screen.getByTestId('loginFlower2')
    const loginEllipseRt = screen.getByTestId('loginEllipseRt')
    const loginEllipseRb = screen.getByTestId('loginEllipseRb')
    const loginEllipseLb = screen.getByTestId('loginEllipseLb')
    const loginEllipseLt = screen.getByTestId('loginEllipseLt')
    const loginSignature = screen.getByTestId('loginSignature')
    expect(loginFlower1).toBeInTheDocument()
    expect(loginFlower2).toBeInTheDocument()
    expect(loginEllipseRt).toBeInTheDocument()
    expect(loginEllipseRb).toBeInTheDocument()
    expect(loginEllipseLb).toBeInTheDocument()
    expect(loginEllipseLt).toBeInTheDocument()
    expect(loginSignature).toBeInTheDocument()
  })

  it('should match snapshoot', () => {
    const { container } = render(<LoginImages />)
    expect(container).toMatchSnapshot()
  })
})
