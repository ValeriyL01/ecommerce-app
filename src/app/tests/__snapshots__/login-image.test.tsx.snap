// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Registration-images should match snapshoot 1`] = `
<div>
  <img
    alt=""
    class="absolute right-0 top-0 z-10 w-1/4 object-cover"
    data-testid="loginFlower1"
    src="/images/loginPageImg/loginPage-flower1.png"
  />
  <img
    alt=""
    class="absolute bottom-0 left-0 z-10 w-1/4 object-cover"
    data-testid="loginFlower2"
    src="/images/loginPageImg/loginPage-flower2.png"
  />
  <img
    alt=""
    class=" z-5 absolute right-0 top-0  w-1/4 object-cover opacity-60"
    data-testid="loginEllipseRt"
    src="/images/loginPageImg/Ellipse rt.png"
  />
  <img
    alt=""
    class=" z-5 absolute bottom-0 left-0  w-1/4 object-cover opacity-60"
    data-testid="loginEllipseLb"
    src="/images/loginPageImg/Ellipse lb.png"
  />
  <img
    alt=""
    class=" z-5 absolute left-0 top-0  w-2/12 object-cover opacity-60"
    data-testid="loginEllipseLt"
    src="/images/loginPageImg/Ellipse lt.png"
  />
  <img
    alt=""
    class=" z-5 absolute bottom-0 right-0  w-2/12 object-cover opacity-60"
    data-testid="loginEllipseRb"
    src="/images/loginPageImg/Ellipse rb.png"
  />
  <img
    alt=""
    class=" z-5 absolute left-1/2 top-1/2 w-[250px]  object-cover "
    data-testid="loginSignature"
    src="/images/loginPageImg/signature.png"
  />
</div>
`;
